geth/jail/jail_cell_test.go

```diff
@@ -1,3 +1,135 @@
-package jail
+package jail_test
 
-// TODO(tiabc): Move appropriate tests from jail_test.go here.
+import (
+	"time"
+
+	"github.com/robertkrimen/otto"
+	"github.com/status-im/status-go/geth/params"
+)
+
+func (s *JailTestSuite) TestJailTimeoutFailure() {
+	require := s.Require()
+	require.NotNil(s.jail)
+
+	newCell, err := s.jail.NewJailCell(testChatID)
+	require.NoError(err)
+	require.NotNil(newCell)
+
+	// Attempt to run a timeout string against a JailCell.
+	_, err = newCell.RunOnLoop(`
+		setTimeout(function(n){
+			if(Date.now() - n < 50){
+				throw new Error("Timedout early");
+			}
+
+			return n;
+		}, 30, Date.now());
+	`)
+
+	require.NotNil(err)
+}
+
+func (s *JailTestSuite) TestJailTimeout() {
+	require := s.Require()
+	require.NotNil(s.jail)
+
+	newCell, err := s.jail.NewJailCell(testChatID)
+	require.NoError(err)
+	require.NotNil(newCell)
+
+	// Attempt to run a timeout string against a JailCell.
+	res, err := newCell.RunOnLoop(`
+		setTimeout(function(n){
+			if(Date.now() - n < 50){
+				throw new Error("Timedout early");
+			}
+
+			return n;
+		}, 50, Date.now());
+	`)
+
+	require.NoError(err)
+	require.NotNil(res)
+}
+
+// TODO(influx6): JCell.Fetch is not yet being used or needed by
+// other API, keep test out for the main time, if method is later required
+// add to common.JailCell interface then uncomment test.
+// func (s *JailTestSuite) TestJailFetch() {
```

Comment:
If we implemented fetch, we must write a test for it. Please, uncomment. There has already been a request from one person from the community regarding `fetch` so it may immediately be used after the release.